patches/net/minecraft/world/effect/MobEffectInstance.java.patch

```diff
@@ -86,16 +86,20 @@
                              .apply(p_324063_, MobEffectInstance.Details::create)
                  )
          );
-+        // TODO: https://github.com/neoforged/NeoForge/issues/986 - Fix this to have vanilla-compatible sync.
-         public static final StreamCodec<ByteBuf, MobEffectInstance.Details> STREAM_CODEC = StreamCodec.recursive(
-             p_329990_ -> StreamCodec.composite(
+-        public static final StreamCodec<ByteBuf, MobEffectInstance.Details> STREAM_CODEC = StreamCodec.recursive(
+-            p_329990_ -> StreamCodec.composite(
++        public static final StreamCodec<RegistryFriendlyByteBuf, MobEffectInstance.Details> STREAM_CODEC = StreamCodec.recursive(
++            p_329990_ -> net.neoforged.neoforge.network.codec.NeoForgeStreamCodecs.composite(
                      ByteBufCodecs.VAR_INT,
-@@ -427,14 +_,23 @@
+                     MobEffectInstance.Details::amplifier,
+                     ByteBufCodecs.VAR_INT,
+@@ -427,14 +_,17 @@
                      MobEffectInstance.Details::showIcon,
                      p_329990_.apply(ByteBufCodecs::optional),
                      MobEffectInstance.Details::hiddenEffect,
-+                    // net.neoforged.neoforge.common.EffectCure.STREAM_CODEC.<java.util.Set<net.neoforged.neoforge.common.EffectCure>>apply(ByteBufCodecs.collection(java.util.HashSet::new)).apply(ByteBufCodecs::optional),
-+                    // MobEffectInstance.Details::cures,
++                    // Neo: Add additional serialization logic for custom EffectCure(s)
++                    net.neoforged.neoforge.network.codec.NeoForgeStreamCodecs.connectionAware(ByteBufCodecs.optional(net.neoforged.neoforge.common.EffectCure.STREAM_CODEC.apply(ByteBufCodecs.collection(java.util.HashSet::new))), net.neoforged.neoforge.network.codec.NeoForgeStreamCodecs.uncheckedUnit(Optional.empty())),
```

Comment:
Please put the two arguments given to `NeoForgeStreamCodecs.connectionAware()` on separate lines, this line is absurdly long.